src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -718,6 +718,14 @@ emit_hardware_intrinsics (
 		if (!info)
 			goto support_probe_complete;
 		id = info->id;
+
+#ifdef TARGET_ARM64
```

Comment:
I'm not sure this is an arm-only problem.

Reply:
We'll just do this for arm64 for now.  It's unclear that we can repro the issue on x64 given that some of the intrinsics have JIT support or they're full AOT. (for example maccatalyst-x64 is an obvious candidate)